web/src/group_permission_settings.ts

```diff
@@ -230,3 +233,82 @@ export function get_assigned_permission_object(
     // The group does not have permission.
     return undefined;
 }
+
+export function check_group_permission_settings_data(): void {
+    const all_realm_group_settings = z
+        .array(realm_group_setting_name_schema)
+        .parse(Object.keys(realm.server_supported_permission_settings.realm));
+    const realm_group_settings_with_subsection_data: RealmGroupSettingName[] = [];
+    for (const subsection_obj of settings_config.realm_group_permission_settings) {
+        for (const setting_name of subsection_obj.settings) {
+            realm_group_settings_with_subsection_data.push(setting_name);
+        }
+    }
+
+    if (
+        !all_realm_group_settings.every((setting_name) =>
+            realm_group_settings_with_subsection_data.includes(setting_name),
+        )
+    ) {
+        blueslip.error("Settings missing in 'settings_config.realm_group_permission_settings'");
```

Comment:
Not sure about wording of the error message. Also, might be helpful to mention setting names.